src/java.base/share/classes/java/lang/String.java

```diff
@@ -2156,6 +2156,10 @@ public boolean regionMatches(int toffset, String other, int ooffset, int len) {
              (ooffset > (long)other.length() - len)) {
             return false;
         }
+        // Any strings match if len < 0
+        if (len < 0) {
+           return true;
+        }
```

Comment:
Isn't it also true that the regions trivial match if len == 0?

Reply:
Yes, you are right. The statement "there is some nonnegative integer k less than len " covers the case len == 0, and the rest of the cases are checked before this line.